contracts/utils/Giver.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+
+import "@yield-protocol/vault-interfaces/src/ICauldron.sol";
+import "@yield-protocol/vault-interfaces/src/DataTypes.sol";
+import "@yield-protocol/utils-v2/contracts/access/AccessControl.sol";
+
+/// @title A contract that allows owner of a vault to give the vault
+contract Giver is AccessControl {
+    mapping(bytes6 => bool) public ilkBlacklist;
+    ICauldron public immutable cauldron;
```

Comment:
For code style, please put immutables and constants at the top of the state variable section.